### ISSUE TITLE
The entry heading of changelog.txt doesn't match the WPORG format, resulting in incorrect SVN version tagging when releasing

### ISSUE THREAD

**Issue by eason9487** [woocommerce/facebook-for-woocommerce#2087]
## 🔎 Isolate the bug

- [x] I have confirmed this occurs in the most recent version of WordPress, WooCommerce, and Facebook for WooCommerce.
- [ ] (N/A) ~~I have confirmed this occurs when only WooCommerce and Facebook for WooCommerce are active and when using a default WordPress or WooCommerce theme.~~

### This issue has probably existed since at least version 2.6.1.

After releasing a new version through the `woorelease` tool, although the log shows success for the deployment of WPORG, the generated URL of extension archive file is missing the version suffix. It can be downloaded normally from WPORG, but it will cause a **404** error when downloading from WCCOM. For example https://downloads.wordpress.org/plugin/facebook-for-woocommerce.zip

### Related error events reporting in Slack

- p1631785700264800-slack-CK365S85V
- p1630430572083900-slack-CK365S85V
- p1624860422398400-slack-C4E0DVDB2

## 🚶‍♀️ Steps to reproduce

Follow the current release steps in [wiki/Build-&-Release](https://github.com/woocommerce/facebook-for-woocommerce/wiki/Build-&-Release). But it's **NOT** recommended trying to reproduce it directly.

## ✔️ Expected behavior

The download URL of extension archive file should contain the released version in the filename. For example https://downloads.wordpress.org/plugin/facebook-for-woocommerce.2.6.5.zip

### The clues

1. From the `woorelease` log, it can be found that there is a problem when parsing changelog data.
    ```
    WR [08:54:02] [NOTICE] Found product slug for WordPress.org: 'facebook-for-woocommerce'. Following .org release process.
    WR [08:54:02] [NOTICE] Parsing changelog data...
    WR [08:54:02] [NOTICE] Extension 'facebook-for-woocommerce' does not contain a valid changelog entry. Try to add generated changelog.
    ```
1. When releasing with the below command, [the specified `product_version` isn't passed to the `wporg:release` command](https://github.com/woocommerce/woorelease/blob/2.0.4/includes/command/class-release.php#L161-L176).
    ```
    wr release --product_version=X.X.X https://github.com/woocommerce/facebook-for-woocommerce/tree/release/X.X.X
    ```
1. The `wporg:release` command would [try to parse a version from the changelog.txt file](https://github.com/woocommerce/woorelease/blob/2.0.4/includes/command/class-wporg-release.php#L84-L88) if the version parameter is empty.
1. The changelog parser looks for the latest version according to whether the product exists on WPORG, but the current changelog format doesn't match [the WPORG pattern](https://github.com/woocommerce/woorelease/blob/2.0.4/includes/tools/class-changelog.php#L57-L58).
    https://github.com/woocommerce/facebook-for-woocommerce/blob/f4e8c6097d6622059705df6d7cbab9179609c055/changelog.txt#L3-L4
1.  Ultimately, it results in [a `null` version parameter](https://github.com/woocommerce/woorelease/blob/2.0.4/includes/tools/class-extension-deploy.php#L63-L73) being called to the `Extension_Deploy::maybe_deploy_wporg()` method.

### Possible solution

Not sure why the `woorelease` doesn't pass down the `product_version` parameter and relies on the fallback of parsing version from changelog, but I think we could fix this issue by revising the format of entry heading of changelog to the WPORG pattern.

**Comment by jconroy** [comment]
@eason9487 can you check/confirm if we can close this one

**Comment by eason9487** [comment]
@jconroy 

> can you check/confirm if we can close this one

Yes, we can close this issue. It has been fixed since `woorelease` 2.1.0
